File: src/design_patterns/behavioral/state/javascript/state.js
class State {
    handle(context) { }
}

class ConcreteStateA extends State {
    handle(context) {
        console.log("Handling request with ConcreteStateA");
        context.state = new ConcreteStateB();
    }
}

class ConcreteStateB extends State {
    handle(context) {
        console.log("Handling request with ConcreteStateB");
        context.state = new ConcreteStateA();
    }
}

class Context {
    constructor() {
        this.state = new ConcreteStateA();
    }

    request() {
        this.state.handle(this);
    }
}

File: src/design_patterns/behavioral/state/python/state.py
class State:
    """ """
    def handle(self, context):
        """

        :param context: 

        """
        pass


class ConcreteStateA(State):
    """ """
    def handle(self, context):
        """

        :param context: 

        """
        print("Handling request with ConcreteStateA")
        context.state = ConcreteStateB()


class ConcreteStateB(State):
    """ """
    def handle(self, context):
        """

        :param context: 

        """
        print("Handling request with ConcreteStateB")
        context.state = ConcreteStateA()


class Context:
    """ """
    def __init__(self):
        self.state = ConcreteStateA()

    def request(self):
        """ """
        self.state.handle(self)
